Source/WebCore/platform/audio/cocoa/AudioFileReaderCocoa.cpp

```diff
@@ -360,11 +361,14 @@ std::optional<size_t> AudioFileReader::decodeWebMData(AudioBufferList& bufferLis
             // So we set it to what there is left to decode instead.
             UInt32 numFrames = std::min<uint32_t>(std::numeric_limits<int32_t>::max() / sizeof(float), numberOfFrames - totalDecodedFrames);
 
-            for (UInt32 i = 0; i < inFormat.mChannelsPerFrame; i++) {
-                decodedBufferList->mBuffers[i].mNumberChannels = 1;
-                decodedBufferList->mBuffers[i].mDataByteSize = numFrames * sizeof(float);
-                decodedBufferList->mBuffers[i].mData = static_cast<float*>(bufferList.mBuffers[i].mData) + decodedFrames;
+            auto decodedBuffers = WebCore::span(*decodedBufferList);
+            auto bufferListBuffers = WebCore::span(bufferList);
+            for (UInt32 i = 0; i < inFormat.mChannelsPerFrame; ++i) {
+                decodedBuffers[i].mNumberChannels = 1;
+                decodedBuffers[i].mDataByteSize = numFrames * sizeof(float);
+                decodedBuffers[i].mData = dataMutableFloatSpan(bufferListBuffers[i]).subspan(decodedFrames).data();
```

Comment:
Maybe this was discussed in past review, but I think `dataMutableFloatSpan` would read better as `mutableDataFloatSpan`. We could also drop the 'data', since no other span is possible: `mutableFloatSpan`.